src/vivarium/framework/engine.py

```diff
@@ -110,6 +108,10 @@ def __init__(self, model_specification: Union[str, Path, ConfigTree] = None,
 
         self.add_components(components)
 
+    @property
+    def name(self):
```

Comment:
This, and all other naming fixes, is to make the constraint maker in the lifecycle system happy.  I was still getting random failures in doctest related to identifying constrained methods due to python garbage collecting population views and lookup tables.  Methods that need to be constrained now must be attached to an object with a unique name.  This is already a deal for our components.  It's now part of, e.g., the context and things like lookup tables, randomness streams, and population views.